lib/node/runner/api/transaction.go

```diff
@@ -136,3 +137,62 @@ func (api NetworkHandlerAPI) GetTransactionsByAccountHandler(w http.ResponseWrit
 	list := p.ResourceList(txs, cursor)
 	httputils.MustWriteJSON(w, 200, list)
 }
+
+func (api NetworkHandlerAPI) GetTransactionStatusByHashHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	key := vars["id"]
+
+	readFunc := func() (payload interface{}, err error) {
+		status := "notfound"
+		if found, _ := block.ExistsTransactionPool(api.storage, key); found {
+			status = "submitted"
+		} else if found, _ = block.ExistsBlockTransaction(api.storage, key); found {
+			status = "confirmed"
+		}
+		payload = resource.NewTransactionStatus(key, status)
+		return payload, nil
+	}
+
+	if httputils.IsEventStream(r) {
+		event := fmt.Sprintf("hash-%s", key)
```

Comment:
@kfangw This code will work fine. One thing suggestion. The event stream for this case, checking status of transaction, will be only meaningful when the transaction is 'not found'. So this block can be rewrite like,

* check the transaction is exists in pool or block transaction,
* if found, respond it.
* if not found and the request is event stream, watch observer.
* if not found and not event stream, just 404.

Reply:
good idea. I will patch it.